From b303e792c3084c0b5535ec58fbae84a5fd4b892b Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Fri, 13 Oct 2017 11:56:48 +0200
Subject: [PATCH 17/69] Disable vhost-user-scsi and vhost-user-scsi-pci

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
Message-id: <20171013115649.20817-2-mrezanin@redhat.com>
Patchwork-id: 77263
O-Subject: [RHV7.5 qemu-kvm-rhev PATCHv2 1/2] Disable vhost-user-scsi and vhost-user-scsi-pci
Bugzilla: 1498496
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

From: Miroslav Rezanina <mrezanin@redhat.com>

We are not going to support these new devices. CONFIG_VHOST_USER_SCSI option
removes only vhost-user-scsi so we have to manually disable
vhost-user-scsi-pci.

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 default-configs/aarch64-softmmu.mak | 2 +-
 default-configs/pci.mak             | 2 +-
 default-configs/ppc64-softmmu.mak   | 2 +-
 default-configs/s390x-softmmu.mak   | 2 +-
 hw/virtio/virtio-pci.c              | 2 ++
 5 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/default-configs/aarch64-softmmu.mak b/default-configs/aarch64-softmmu.mak
index 5d57303..a2d05f4 100644
--- a/default-configs/aarch64-softmmu.mak
+++ b/default-configs/aarch64-softmmu.mak
@@ -22,7 +22,7 @@ CONFIG_SMBIOS=y
 CONFIG_PL061=y
 CONFIG_GPIO_KEY=y
 CONFIG_ARM_V7M=y
-CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
+#CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
 CONFIG_PCIE_PORT=y
 CONFIG_XIO3130=y
 CONFIG_IOH3420=y
diff --git a/default-configs/pci.mak b/default-configs/pci.mak
index 9bd8452..8cf0b60 100644
--- a/default-configs/pci.mak
+++ b/default-configs/pci.mak
@@ -43,4 +43,4 @@ CONFIG_VGA=y
 CONFIG_VGA_PCI=y
 CONFIG_IVSHMEM_DEVICE=$(CONFIG_IVSHMEM)
 #CONFIG_ROCKER=y
-CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
+#CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
diff --git a/default-configs/ppc64-softmmu.mak b/default-configs/ppc64-softmmu.mak
index 31ef40c..46e356b 100644
--- a/default-configs/ppc64-softmmu.mak
+++ b/default-configs/ppc64-softmmu.mak
@@ -8,7 +8,7 @@ CONFIG_USB_XHCI=y
 CONFIG_USB_XHCI_NEC=y
 CONFIG_WDT_IB6300ESB=y
 CONFIG_PCI_TESTDEV=y
-CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
+#CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
 
 include sound.mak
 include usb.mak
diff --git a/default-configs/s390x-softmmu.mak b/default-configs/s390x-softmmu.mak
index 2846634..02f62de 100644
--- a/default-configs/s390x-softmmu.mak
+++ b/default-configs/s390x-softmmu.mak
@@ -1,6 +1,6 @@
 CONFIG_PCI=y
 #CONFIG_VIRTIO_PCI=y
-CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
+#CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
 CONFIG_VIRTIO=y
 CONFIG_SCLPCONSOLE=y
 # Disabled for Red Hat Enterprise Linux:
diff --git a/hw/virtio/virtio-pci.c b/hw/virtio/virtio-pci.c
index eccf809..8208aa2 100644
--- a/hw/virtio/virtio-pci.c
+++ b/hw/virtio/virtio-pci.c
@@ -2665,9 +2665,11 @@ static void virtio_pci_register_types(void)
 #ifdef CONFIG_VHOST_SCSI
     type_register_static(&vhost_scsi_pci_info);
 #endif
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 #if defined(CONFIG_VHOST_USER) && defined(CONFIG_LINUX)
     type_register_static(&vhost_user_scsi_pci_info);
 #endif
+#endif
 #ifdef CONFIG_VHOST_VSOCK
     type_register_static(&vhost_vsock_pci_info);
 #endif
-- 
1.8.3.1

